qemu/tests/timedrift_adjust_time.py

```diff
@@ -198,14 +220,28 @@ def check_drift_after_adjust_time(self, session):
         while time.time() < start_time + timeout:
             host_epoch_time, guest_epoch_time = self.get_epoch_seconds(session)
             real_difference = abs(host_epoch_time - guest_epoch_time)
-            if abs(real_difference - expect_difference) < tolerance:
-                return
+            if self.params["os_type"] == 'linux':
+                expect_difference_hwclock = float(self.params["time_difference_hwclock"])
+                guest_hwtime = self.get_hwtime(session)
+                real_difference_hw = abs(host_epoch_time - guest_hwtime)
+                if abs(real_difference - expect_difference) < tolerance or \
+                   abs(real_difference - expect_difference_hwclock) < tolerance:
```

Comment:
I think it should be `and`, both system time and hw time gap should less than the tolerance.  Otherwise, test fail.